wicket-core/src/main/java/org/apache/wicket/protocol/http/WebApplication.java

```diff
@@ -1074,4 +1074,39 @@ public boolean getUpdateAutoLabelsOnAjaxRequests()
 	{
 		return true;
 	}
+
+	/**
+	 * Returns a new web application.
+	 * @param homePageClass Home page class
+	 * @return new web application
+	 */
+	public static WebApplication newWebApp(final Class<? extends Page> homePageClass) {
+		return newWebApp(homePageClass, null);
+	}
+
+	/**
+	 * Returns a new web application.
+	 * @param homePageClass Home page class
+	 * @param initConsumer  Lambda for initializing the web application
+	 * @return new web application
+	 */
+	public static WebApplication newWebApp(final Class<? extends Page> homePageClass, final SerializableConsumer<WebApplication> initConsumer) {
```

Comment:
I don't imagine a good use case where someone would like to use an anonymous instance of WebApplication.
This is useful only for mini applications like the quickstart. But any real application most definitely will need a proper specialization and its own logic.